src/services/superfluidMinionService.js

```diff
@@ -0,0 +1,142 @@
+import Web3 from 'web3';
+
+import SuperfluidMinionAbi from '../contracts/superfluidMinion.json';
+import { SUPERFLUID_MINION_STREAMS } from '../graphQL/superfluid-queries';
+import { TokenService } from './tokenService';
+import { chainByID, getGraphEndpoint } from '../utils/chain';
+import { graphFetchAll } from '../utils/theGraph';
+
+import SuperfluidSDK from '@superfluid-finance/js-sdk';
+
+let sf = null;
+
+const getSuperTokenBalances = async (chainID, minion, superTokens) => {
+  const tokenBalances = superTokens.map(async (tokenAddress) => {
+    const tokenService = TokenService({
+      tokenAddress: tokenAddress,
+      chainID,
+    });
+    return {
+      [tokenAddress]: {
+        tokenBalance: await tokenService('balanceOf')(minion),
+        symbol: await tokenService('symbol')(minion),
+        decimals: await tokenService('decimals')(minion),
+      },
+    };
+  });
+  return Object.assign({}, ...(await Promise.all(tokenBalances)));
+};
+
+export const SuperfluidMinionService = ({ web3, minion, chainID }) => {
+  const chainConfig = chainByID(chainID);
+  if (!web3) {
+    const rpcUrl = chainConfig.rpc_url;
+    web3 = new Web3(new Web3.providers.HttpProvider(rpcUrl));
+  }
+
+  const contract = new web3.eth.Contract(SuperfluidMinionAbi, minion);
+
+  return function getService(service) {
+    if (service === 'fetchStreams') {
+      return async ({ molochAddress }) => {
+        const streams = await graphFetchAll({
```

Comment:
try/catch

Reply:
this will be reviewed again in the next PR update